### ISSUE TITLE
'NoneType' object is not callable error on az functionapp create with --assign-identity

### ISSUE THREAD

**Issue by roycornelissen** [Azure/azure-cli#18116]
## Describe the bug

**Command Name**
`az functionapp create` with `--assign-identity` parameter provided.

Discovered that running e.g. `az webapp identity assign` ... or `az webapp identity remove` yields the same error. Anything related to user assigned identities seems to be broken, so it's not only `az functionapp` related.

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
'NoneType' object is not callable
Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 231, in invoke
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 657, in execute
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 712, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/appservice/commands.py", line 37, in _ex_handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 691, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 328, in __call__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/appservice/custom.py", line 3030, in create_function
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/appservice/custom.py", line 789, in assign_identity
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/arm.py", line 706, in assign_identity
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/appservice/custom.py", line 782, in setter
TypeError: 'NoneType' object is not callable

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

A user assigned identity exists.

- `az functionapp create --resource-group {} --name {} --plan {} --assign-identity {} --storage-account {} --app-insights-key {} --functions-version {}`

## Expected Behavior
Command should succeed. Our deployment script has been working fine with `azure-cli 2.22.1`. I can now consistently reproduce the error with `2.23.0`.

## Environment Summary
```
Windows-10-10.0.19041-SP0
Python 3.8.9
Installer: MSI

azure-cli 2.23.0

Extensions:
aks-preview 0.5.12
azure-devops 0.18.0
managementpartner 0.1.3

```
## Additional Context
Azure CLI 2.22.1 does not have this issue.

**Comment by yonzhan** [comment]
route to service team

**Comment by roycornelissen** [comment]
The exception seems to be occur when `--assign-identity` parameter is provided.

**Comment by roycornelissen** [comment]
Azure CLI 2.22.1 works fine BTW

**Comment by yonzhan** [comment]
@ahmedelnably, @fabiocav, any update for this?

**Comment by fabiocav** [comment]
@apawast can we please flag this for triage?

**Comment by roycornelissen** [comment]
@yonzhan @apawast @fabiocav any update? We're unable to perform any action related to user assigned managed identities with the latest Azure CLI.

**Comment by roycornelissen** [comment]
We just received the new update version `2.24.0` and it fixes the issues with managed identities. Hopefully the Azure build agents are updated soon.

**Comment by gzuber** [comment]
Yes, we checked in a fix to 2.24.0. Sorry for the inconvenience. @yonzhan I think we can close this, thanks!
